lib/Db/TagMapper.php

```diff
@@ -145,20 +145,22 @@ public function getAllTagsForMessages(array $messages): array {
 
 		$qb = $this->db->getQueryBuilder();
 		$idsQuery = $qb->select('mt.*')
+			->addSelect('t.*')
 			->from('mail_message_tags', 'mt')
+			->join('mt', $this->getTableName(), 't', $qb->expr()->eq('mt.tag_id','t.id'))
 			->where(
-				$qb->expr()->in('imap_message_id', $qb->createNamedParameter($ids, IQueryBuilder::PARAM_STR_ARRAY))
+				$qb->expr()->in('mt.imap_message_id', $qb->createNamedParameter($ids, IQueryBuilder::PARAM_STR_ARRAY))
 			);
 		$idsQuery = $idsQuery->execute();
 		$queryResult = $idsQuery->fetchAll();
 		if (empty($queryResult)) {
 			return [];
 		}
-		$result = [];
-		foreach ($queryResult as $qr) {
-			$result[] = $qr['imap_message_id'];
-			$result[$qr['imap_message_id']][] = $this->getTag((int)$qr['tag_id']);
-		};
+		$result = array_map(function ($qr) {
+			$transformed[] = $qr['imap_message_id'];
```

Comment:
this code confused me also in the previous version

so here you push a new item to the end of the array (at the numeric index or whatever php assigns)

Reply:
because i want the array to be indexed by the imap_message_id so I can access it in the message mapper via: 			`$dbTags = $tags[$message->getMessageId()] ?? [];` - otherwise I will have to loop through the `$tags` array to find those entries.